force-app/main/default/classes/SOQL.cls

```diff
@@ -288,20 +333,21 @@ public with sharing class SOQL {
      */
     public static List<Account> findAccountsByLeadEmailDomains(Set<Id> leadIds) {
         // Query the Leads
-
+        List<Lead> myLeads = [SELECT Id, Email FROM Lead WHERE Id IN :leadIds];
         // Create a set of email domains from the Leads
-
         // Loop through each Lead
-
-            // Get the email from the Lead
-
-            // Check if the email is not null
-
-                // Add the email domain to the set
-
-        
+        // Get the email from the Lead
+        // Check if the email is not null
+        // Add the email domain to the set
+        Set<String> emailDoms = new Set<String>();
+        for (Lead l : myLeads) {
+            if(l.Email != null){
+                emailDoms.add('www.'+l.Email.substringAfter('@'));
```

Comment:
This is being pendantic, but in a real scenario, dont assume your users will always put "www." in the beginning.